go/tasks/plugins/webapi/snowflake/plugin.go

```diff
@@ -0,0 +1,286 @@
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	logger.Info(ctx, "Deleted query execution [%v]", resp)
+
+	return nil
+}
+
+func (p Plugin) Status(_ context.Context, taskCtx webapi.StatusContext) (phase core.PhaseInfo, err error) {
+	exec := taskCtx.ResourceMeta().(*ResourceMetaWrapper)
+	status := taskCtx.Resource().(*ResourceWrapper).Status
+	if status == "" {
+		return core.PhaseInfoUndefined, errors.Errorf(ErrSystem, "No Status field set.")
+	}
+	statusCode, err := strconv.Atoi(status)
+	if err != nil {
+		return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", status)
+	}
+	taskInfo := createTaskInfo(exec.QueryID, exec.Account)
+	switch statusCode {
+	case http.StatusAccepted:
+		return core.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, createTaskInfo(exec.QueryID, exec.Account)), nil
+	case http.StatusOK:
+		return pluginsCore.PhaseInfoSuccess(taskInfo), nil
+	case http.StatusUnprocessableEntity:
+		return pluginsCore.PhaseInfoFailure(status, "phaseReason", taskInfo), nil
+	}
+	return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", status)
+}
+
+func buildRequest(method string, queryInfo QueryInfo, account string, token string,
+	queryId string, isCancel bool) (*http.Request, error) {
+	snowflakeUrl := "https://" + account + ".snowflakecomputing.com/api/statements"
+	var data []byte
+	if method == "POST" && !isCancel {
+		snowflakeUrl += "?async=true"
+		data = []byte(fmt.Sprintf(`{
+		  "statement": "%v",
+		  "database": "%v",
+		  "schema": "%v",
+		  "warehouse": "%v"
+		}`, queryInfo.Statement, queryInfo.Database, queryInfo.Schema, queryInfo.Warehouse))
+	} else {
+		snowflakeUrl += "/" + queryId
+	}
+	if isCancel {
+		snowflakeUrl += "/cancel"
+	}
+
+	req, err := http.NewRequest(method, snowflakeUrl, bytes.NewBuffer(data))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", "Bearer "+token)
+	req.Header.Add("X-Snowflake-Authorization-Token-Type", "KEYPAIR_JWT")
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "application/json")
+	return req, nil
+}
+
+func buildResponse(response *http.Response) (map[string]interface{}, error) {
+	responseBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	var data map[string]interface{}
+	err = json.Unmarshal(responseBody, &data)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func createTaskInfo(queryId string, account string) *core.TaskInfo {
+	timeNow := time.Now()
+
+	return &core.TaskInfo{
+		OccurredAt: &timeNow,
+		Logs: []*flyteIdlCore.TaskLog{
+			{
+				Uri: fmt.Sprintf("https://%v.snowflakecomputing.com/console#/monitoring/queries/detail?queryId=%v",
+					account,
+					queryId),
+				Name: "Snowflake Console",
+			},
+		},
+	}
+}
+
+func getSnowflakeToken() string {
+	return os.Getenv("SNOWFLAKE_TOKEN")
+}
```

Comment:
Don't know how to properly get the token. 
1. Get environment variable from k8s?
2. Get from the secret in k8s?
3. Users specify the token in Flytekit?
4. or Others

Any thought?

Reply:
Users should probably not be the ones to specify the token, though perhaps they should be able to specify an override.  we can do that in the future if it becomes necessary.

The plugin should probably fetch the k8s secrets from the setup context https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/setup_context.go#L22

@kumare3 @EngHabu could you point kevin to the docs to all the secrets stuff?

Reply:
https://docs.flyte.org/projects/cookbook/en/latest/auto/core/containerization/use_secrets.html?highlight=secret

Reply:
So the backend secrets are accessed using the
[SecretManager](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/secret_manager.go), which is given in the context [here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/exec_context.go#L32)

For backend this is simply accessing File based secrets using [this](https://github.com/flyteorg/flytepropeller/blob/master/pkg/controller/nodes/task/handler.go#L777)

Here is the code
https://github.com/flyteorg/flytepropeller/blob/c59db1be84897f8465dd14413c65088bce600646/pkg/controller/nodes/task/secretmanager/secrets.go#L25

So you can inject the secret into flytepropeller env, or use k8s secrets to mount a secret into the pod. Preferred to mount it into the pod

Reply:
The secret itself should be added during deployment and it should be done as part of the helm / kustomize manifest

Reply:
cc @EngHabu, we did not merge the paths for the webhook and propeller, i thought we were going to?

Reply:
Thx! I updated it.